File: src/main/scala/com/scala/kafka.stream/model/ProduitBrut.scala
package com.scala.test.model

import java.math.BigDecimal

import play.api.libs.json.{JsNumber, JsValue, Json, Writes}

case class ProduitBrut(
                        id: Option[Long],
                        price: Option[BigDecimal])

object ProduitBrut {
  implicit val bigdecimalWrites = new Writes[BigDecimal] {
    def writes(o: BigDecimal): JsValue = {
      JsNumber(o.setScale(2))
    }
  }
  implicit val format = Json.format[ProduitBrut]
}

File: kafka-stream/src/main/java/com/java/kafka/stream/model/ProduitBrut.java
package com.java.kafka.stream.model;

import java.math.BigDecimal;

public class ProduitBrut {

    private Long id;
    private BigDecimal price;

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return "ProduitBrut{" +
                "id=" + id +
                ", price=" + price +
                '}';
    }
}
